grails-app/services/UserProfileService.groovy

```diff
@@ -0,0 +1,29 @@
+import grails.transaction.Transactional
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.transmart.searchapp.AuthUser
+import grails.converters.JSON
+
+@Transactional
+class UserProfileService {
```

Comment:
All new services and src/groovy classes should use `@CompileStatic` except for cases where the code would be significantly more verbose or complicated.